Assets/Scripts/EOSStatsManager.cs

```diff
@@ -0,0 +1,176 @@
+    /// </summary>
+    public class EOSStatsManager : EOSOnConnectLogin, IEOSSubManager
+    {
+        /// <summary>
+        /// Cache of the map between the product user id and the stats for that user.
+        /// </summary>
+        private Dictionary<ProductUserId, List<Stat>> _productUserIdToStatsCache = new();
+
+        protected override void PostSuccessfulLogin()
+        {
+            QueryStatsForProductUserId(_productUserId, (ref OnQueryStatsCompleteCallbackInfo statsQueryData) =>
+            {
+                CacheStatsForProductUserId(_productUserId);
+            });
+        }
+
+        /// <summary>
+        /// Gets the StatsInterface for EOS
+        /// </summary>
+        /// <returns>The stats interface.</returns>
+        private StatsInterface GetEOSStatsInterface()
+        {
+            return EOSManager.Instance.GetEOSPlatformInterface().GetStatsInterface();
+        }
+
+        private void CacheStatsForProductUserId(ProductUserId productUserId)
+        {
+            var statInterface = GetEOSStatsInterface();
+            GetStatCountOptions countOptions = new()
+            {
+                TargetUserId = productUserId
+            };
+            uint statsCountForProductUserId = statInterface.GetStatsCount(ref countOptions);
+
+            List<Stat> collectedStats = new();
+            var copyStatsByIndexOptions = new CopyStatByIndexOptions
+            {
+                TargetUserId = productUserId,
+                StatIndex = 0
+            };
+
+            for (uint i = 0; i < statsCountForProductUserId; ++i)
+            {
+                copyStatsByIndexOptions.StatIndex = i;
+
+                Result copyStatResult = statInterface.CopyStatByIndex(ref copyStatsByIndexOptions, out Stat? stat);
+
+                if (copyStatResult == Result.Success)
+                {
+                    collectedStats.Add(stat.Value);
+                }
+            }
+
+            _productUserIdToStatsCache[productUserId] = collectedStats;
+        }
+
+        private void QueryStatsForProductUserId(ProductUserId productUserId, OnQueryStatsCompleteCallback callback)
+        {
+            if (!productUserId.IsValid())
+            {
+                print("Invalid product user id sent in!");
+                return;
+            }
+            var statInterface = GetEOSStatsInterface();
+
+            QueryStatsOptions statsOptions = new()
+            {
+                LocalUserId = productUserId,
+                TargetUserId = productUserId
+            };
+
+            statInterface.QueryStats(ref statsOptions, null, (ref OnQueryStatsCompleteCallbackInfo queryStatsCompleteCallbackInfo) =>
+            {
+                if (queryStatsCompleteCallbackInfo.ResultCode != Result.Success)
+                {
+                    // TODO: handle error
+                    print("Failed to query stats: " + queryStatsCompleteCallbackInfo.ResultCode);
```

Comment:
Similar above; can we run the callback function provided passing in a struct indicating failure?

Reply:
While I am a fan of doing something like that - this PR has the goal of just transplanting the functionality that exists within `EOSAchievementsManager` that pertain to the `Stats Interface` as Epic defines it. Which means expanding upon, or altering the implementation should be considered out of scope for this PR - since these lines were copied over.